src/crypto/README.md

```diff
@@ -112,12 +112,15 @@ the `ByteSource::Builder` without releasing it as a `ByteSource`.
 
 ### `ArrayBufferOrViewContents`
 
-The `ArrayBufferOfViewContents` class is a helper utility that abstracts
+The `ArrayBufferOrViewContents` class is a helper utility that abstracts
 `ArrayBuffer`, `TypedArray`, or `DataView` inputs and provides access to
 their underlying data pointers. It is used extensively through `src/crypto`
 to make it easier to deal with inputs that allow any `ArrayBuffer`-backed
 object.
 
+The lifetime of `ArrayBufferOrViewContents` should not exceed the
+lifetime of its input.
```

Comment:
This changes the API contract in a way that’s easy to get wrong. Can we add restrictions similar to these for `HandleScope` to prevent accidental misuse?

https://github.com/nodejs/node/blob/7f7a899fa5f3b192d4f503f6602f24f7ff4ec57a/deps/v8/include/v8-local-handle.h#L92-L93

https://github.com/nodejs/node/blob/7f7a899fa5f3b192d4f503f6602f24f7ff4ec57a/deps/v8/include/v8-local-handle.h#L104-L109

Reply:
Thanks; I will make these changes! Do you think that we should add similar changes to [`ArrayBufferViewContents`](https://github.com/nodejs/node/blob/a7e5b413efd5f98d4b9acf39ff5e87ddbd8c63ff/src/util.h#L493-L497), which has similar lifetime semantics?

Reply:
Done - I had to make some minor changes because we were using the copy constructor already.

Reply:
> Do you think that we should add similar changes to [`ArrayBufferViewContents`](https://github.com/nodejs/node/blob/a7e5b413efd5f98d4b9acf39ff5e87ddbd8c63ff/src/util.h#L493-L497), which has similar lifetime semantics? (In a separate PR, of course.)

Yes – thanks for bringing this up! 👍